comp/dogstatsd/serverDebug/debug_test.go

```diff
@@ -30,6 +31,7 @@ func fulfillDeps(t testing.TB) Component {
 }
 
 func TestDebugStatsSpike(t *testing.T) {
+	config.Datadog.Set("dogstatsd_logging_enabled", false)
```

Comment:
Similar to my other comment, you will also need to use the injected component here too. You can update `fulfillDeps` [to something similar to what we do here](https://github.com/DataDog/datadog-agent/blob/main/comp/dogstatsd/server/server_test.go#L63)

And then provide your config override as an argument. The benefit of this new interface is you no longe have to `defer unset` the config anymore :)